tests/bgp/test_bgp_bbr.py

```diff
@@ -153,23 +154,41 @@ def config_bbr_enabled(duthosts, setup, rand_one_dut_hostname, restore_bbr_defau
     enable_bbr(duthost, setup['tor1_namespace'])
 
 
+def get_bbr_default_state(duthost):
+    bbr_supported = False
+    bbr_default_state = 'disabled'
+
+    # Check BBR configuration from config_db first
+    bbr_config_db_exist = int(duthost.shell('redis-cli -n 4 HEXISTS "BGP_BBR|all" "status"')["stdout"])
+    if bbr_config_db_exist:
+        # key exist, BBR is supported
+        bbr_supported = True
+        bbr_default_state = duthost.shell('redis-cli -n 4 HGET "BGP_BBR|all" "status"')["stdout"]
+    else:
+        # Check BBR configuration from constants.yml
+        constants = yaml.safe_load(duthost.shell('cat {}'.format(CONSTANTS_FILE))['stdout'])
+        try:
+            bbr_supported = constants['constants']['bgp']['bbr']['enabled']
+            if not bbr_supported:
+                return bbr_supported, bbr_default_state
+            bbr_default_state = constants['constants']['bgp']['bbr']['default_state']
+        except KeyError:
+            return bbr_supported, bbr_default_state
+
+    return bbr_supported, bbr_default_state
+
+
 @pytest.fixture(scope='module')
 def setup(duthosts, rand_one_dut_hostname, tbinfo, nbrhosts):
     duthost = duthosts[rand_one_dut_hostname]
 
     constants_stat = duthost.stat(path=CONSTANTS_FILE)
     if not constants_stat['stat']['exists']:
-        pytest.skip('No file {} on DUT, BBR is not supported')
+        pytest.skip('No constants.yml file on DUT, BBR is not supported')
```

Comment:
If there is no constants.yml, is it a correct status? In my understanding, constants.yml should store many constans, right? If the file doesn't exist, it is in fatal condition, not just BBR is not supported, we have to fail the case, not just skip it, is it correct?

Reply:
I don't think constants file not exist should fail the case, it's not the bbr case's responsibility to make sure the file exists.